packages/beacon-node/src/chain/archiveStore/observers/frequentStateArchiveObserver.ts

```diff
@@ -0,0 +1,20 @@
+import {CheckpointWithHex} from "@lodestar/fork-choice";
+import {QueueObserver} from "../../../system.js";
+import {PROCESS_FINALIZED_CHECKPOINT_QUEUE_LEN} from "../constants.js";
+import {FrequentStateArchiveModules, maybeArchiveState} from "../utils/frequentStateArchive.js";
+
+export class FrequentStateArchiveObserver extends QueueObserver {
+  protected archiveStateEpochFrequency: number;
+
+  constructor(
+    protected modules: FrequentStateArchiveModules,
+    {signal, archiveStateEpochFrequency}: {signal: AbortSignal; archiveStateEpochFrequency: number}
+  ) {
+    super({logger: modules.logger, maxQueueLength: PROCESS_FINALIZED_CHECKPOINT_QUEUE_LEN, signal});
+    this.archiveStateEpochFrequency = archiveStateEpochFrequency;
+  }
+
+  async onForkChoiceFinalized(finalized: CheckpointWithHex): Promise<void> {
+    await maybeArchiveState(this.modules, {archiveStateEpochFrequency: this.archiveStateEpochFrequency}, finalized);
```

Comment:
Do we still need to accommodate the timing between this and the ArchiveBlocksTask that was mentioned in the comment here:

```
      // should be after ArchiveBlocksTask to handle restart cleanly
      await this.statesArchiverStrategy.maybeArchiveState(finalized, this.metrics);
```

Looks like the ArchiveBlocksTask moved to its own observer?  The bigger question is that with several different listeners, how do we accommodate having the tasks sequenced in a specific way?  Are there other conditions that may be dependent on timing of the tasks?